include/core/physical_property_model.h

```diff
@@ -36,15 +36,29 @@ enum field : int
   previous_temperature
 };
 
+inline void
+set_field_vector(const field                          id,
+                 const std::vector<double> &          data,
+                 std::map<field, std::vector<double>> fields)
+{
+  std::vector<double> &target = fields.at(id);
+  size_t               sz     = target.size();
+  for (size_t i = 0; i < sz; ++i)
+    {
+      target[i] = data[i];
```

Comment:
The std::vector class has it's own copy constructor. It could look something like:
`std::vector<double> target(fields[id]); `

Reply:
Yeah but here we don't want to allocate a new vector right, just to copy the data into it.We want to avoid dynamic allocation of vectors when assembling the matrices.